### PR TITLE
pkg/openwsn: update doc

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR updates OpenWSN documentation which was quite out of sync.

### Testing procedure

Read the docs.

### CODE REVIEW

**Comment by maribu** [comment]
The doxygen references don't resolve. Care to fix?

**Review by benpicco — APPROVED** [review]

File: pkg/openwsn/doc.txt [hunk 1]
```diff
@@ -102,30 +103,46 @@
 
  ### radio
 
- The radio adaptation runs in its own thread with the highest priority
- (`THREAD_PRIORITY_MAIN - 4`) and maps to RIOT's @ref netdev API.
+ The radio adaptation maps to RIOT's @ref netdev API or on
+ @ref drivers_ieee802154_hal API. The later is preferred but not yet
+ supported by all BOARDs.
```
Comment (aabadie) [inline pkg/openwsn/doc.txt:108]:
```suggestion
 supported by all boards.
```

File: pkg/openwsn/doc.txt [hunk 2]
```diff
@@ -102,30 +103,46 @@
...
+
+ #### netdev
+
  The radio adaptation preloads the buffer so `NETOPT_PRELOADING` must be
  supported.
 
- OpenWSN needs to be notified when a frame reception/transmission starts and
- when it ends. Therefore radio drivers need to support the following netdev
- events:
+ The radio drivers should to support the following netdev events:
```
Comment (aabadie) [inline pkg/openwsn/doc.txt:139]:
```suggestion
 The radio drivers should support the following netdev events:
```

**Comment by fjmolinas** [comment]
Thanks for the review!

**Pull request merged**